### ISSUE TITLE
Upgrade to TypeScript 4.2

### ISSUE THREAD

**Issue by kevinmarrec** [nuxt/typescript#489]
Microsoft just announced **TypeScript 4.2** :

https://devblogs.microsoft.com/typescript/announcing-typescript-4-2/

Let's upgrade to it, but we need first `@typescript-eslint/typescript-estree` to support it, to prevent this warning :

```
WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <4.2.0

YOUR TYPESCRIPT VERSION: 4.2.2

Please only submit bug reports when using the officially supported version.
```

**Comment by climba03003** [comment]
`TypeScript 4.2` is currently supported by `@typescript-eslint/typescript-estree`
Related Commit: https://github.com/typescript-eslint/typescript-eslint/pull/3112
Supported Version: >= 4.16.0
